docs/Using Randomness/chainlink-vrf.md

```diff
@@ -2,143 +2,77 @@
+- [Wrapper](/docs/vrf/v2/wrapper/): Very similar to [Connect ANY API](/docs/request-and-receive-data/): Consuming contracts directly pay with LINK when they request for randomness. Hence users have to directly fund their consumer contracts and ensure there are enough LINK tokens to pay for randomness requests.
+
+## Choosing the right method
+
+Depending on your use case, one method might be more suitable than another. The recommendations are the following:
+
+- If your use case requires to regularly request for randomness then choose the subscription method. Otherwise , choose the wrapper method. The wrapper method is more suitable to one-off/seldom requests.
+- If you have several VRF consumer contracts then choose the subscription method.
+- If you want more control on the maximum gas price to be paid and less gas overhead then choose the subscription method. You can read the [Subscription](/docs/vrf/v2/subscription/) and [Wrapper](/docs/vrf/v2/wrapper/) explanation pages to understand how the transaction costs are calculated.
+- Due the wrapper overhead, you can request up to 10 random words in a single request. While with the subscription method, you can request up to 200 random words depending on the network.
+
 ## Supported networks
 
 Chainlink VRF v2 is currently available on the following networks:
 
 - Ethereum:
-  - [Mainnet](/docs/vrf-contracts/#ethereum-mainnet)
-  - [Rinkeby testnet](/docs/vrf-contracts/#rinkeby-testnet)
+  - Mainnet
+  - Rinkeby testnet
 - BNB Chain:
-  - [Mainnet](/docs/vrf-contracts/#bnb-chain)
-  - [Testnet](/docs/vrf-contracts/#bnb-chain-testnet)
+  - Mainnet
+  - Testnet
 - Polygon (Matic):
-  - [Mainnet](/docs/vrf-contracts/#polygon-matic-mainnet)
-  - [Mumbai Testnet](/docs/vrf-contracts/#polygon-matic-mumbai-testnet)
+  - Mainnet
+  - Mumbai Testnet
 - Avalanche:
-  - [Avalanche Mainnet](/docs/vrf-contracts/#avalanche-mainnet)
-  - [Avalanche Fuji Testnet](/docs/vrf-contracts/#avalanche-fuji-testnet)
+  - Avalanche Mainnet
+  - Avalanche Fuji Testnet
 - Fantom:
-  - [Fantom Mainnet](/docs/vrf-contracts/#fantom-mainnet)
-  - [Fantom Testnet](/docs/vrf-contracts/#fantom-testnet)
+  - Fantom Mainnet
+  - Fantom Testnet
 
-See the [Contract Addresses](/docs/vrf-contracts) page for a complete list of coordinator addresses and gas price limits.
+Depending on the method method that you choose: See the [Subscription Configuration](/docs/vrf/v2/subscription/configuration/) or the [Wrapper Configuration](/docs/vrf/v2/wrapper/configuration/) pages for a complete list of contract addresses and gas price limits.
```

Comment:
You might move this above the list of networks just so the links are easier to find.

Reply:
done in next push